src/core/dbus-path.c

```diff
@@ -113,6 +113,7 @@ static int bus_path_set_transient_property(
                                 s->path = TAKE_PTR(k);
                                 s->type = t;
                                 s->inotify_fd = -1;
+                                s->old_inotify_fd = -1;
```

Comment:
not sure i grok this commit... it should be entirely fine if we lose events. what's key is that when coming back we install the inotify watch first, and then check whether the file exists in the moment, and act on that... it doesn't matter if the file was modified x times while we were out, it matters if it exists afterwards

Reply:
So... is PathExists= supposed to be an edge-trigger or a level-trigger?
The man page implies the former:
> If the file specified exists, the configured unit is activated.

And behaviour (before this PR) matches this too: the unit is *not* restarted automatically whenever it exits. It is restarted automatically when daemon-reload is executed, but that seems like a bug because daemon-reload should not have of restarting arbitrary units. And making the units auto-restart seems problematic: I assume people often use path triggering with short-lived services that process the file and exit. An occasional spurious restart is just a slight annoyance, but restarting in a tight loop would be problematic. So even from backwards-compat perspective we can't make PathExists= level-triggered.

If PathExists= is edge-triggered, and need to be careful when doing the restart not to miss events. Just checking after the restart is not enough, as described in the commit message.

Reply:
level triggered

i.e. whenver the file exist we start the unit unless it is already started.

Reply:
And we keep restarting it when it exits?